### PR TITLE
Java: File IO skill

### PR DESCRIPTION
**Pull request by brunoborges**

### CODE REVIEW

**Comment by brunoborges** [comment]
@markwallace-microsoft problem solved.

File: java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/FileIOSkill.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+     * @param path Source file.
+     * @return File content.
+     */
+    @DefineSKFunction(description = "Read a file", name = "readAsync")
+    public Mono<String> readFileAsync(
+            @SKFunctionInputAttribute
+                    @SKFunctionParameters(name = "path", description = "Source file")
+                    String path,
+            SKContext context) {
+        return Mono.create(
```
Comment (johnoliver) [inline java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/FileIOSkill.java:48]: I would probably not have an async method here, just return String and let the framework wrap it

File: java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/FileIOSkill.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                            description = "File content",
+                            defaultValue = "",
+                            type = String.class)
+                    String content,
+            SKContext context) {
+        return Mono.create(
+                monoSink -> {
+                    try {
+                        Path filePath = Paths.get(path);
+                        String charset = context.getVariables().get("charset");
```
Comment (johnoliver) [inline java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/FileIOSkill.java:92]: can take charset as an argument

File: FEATURE_MATRIX.md [hunk 3]
```diff
@@ -90,4 +90,4 @@ in the interest of velocity and feature parity. The project is now going through
 a refactoring exercise to increase code quality.
 
 To make the SDK as lightweight as possible, the core packages have
-a minimal set of external dependencies.
```
Comment (markwallace-microsoft) [inline FEATURE_MATRIX.md:93]: Can you remove this file from the PR

**Review by markwallace-microsoft — CHANGES_REQUESTED** [review]

**Review by johnoliver — APPROVED** [review]

**Pull request merged**